crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -794,162 +809,252 @@ where
         None
     };
 
-    let mut block_importing_notification_stream = subspace_link
+    let block_importing_notification_stream = subspace_link
         .block_importing_notification_stream
         .subscribe();
 
     Ok(async move {
-        let archiver = match maybe_archiver {
-            Some(archiver) => archiver,
-            None => initialize_archiver(&segment_headers_store, &subspace_link, client.as_ref())?,
-        };
+        let mut saved_block_import_notification: Option<BlockImportingNotification<Block>> = None;
 
-        let InitializedArchiver {
-            confirmation_depth_k,
-            mut archiver,
-            older_archived_segments,
-            best_archived_block: (mut best_archived_block_hash, mut best_archived_block_number),
-        } = archiver;
-
-        let archived_segment_notification_sender =
-            subspace_link.archived_segment_notification_sender.clone();
-
-        // Farmers may have not received all previous segments, send them now.
-        for archived_segment in older_archived_segments {
-            send_archived_segment_notification(
-                &archived_segment_notification_sender,
-                archived_segment,
-            )
-            .await;
-        }
+        let mut block_importing_notification_stream = block_importing_notification_stream;
 
-        while let Some(BlockImportingNotification {
-            block_number,
-            // Just to be very explicit that block import shouldn't continue until archiving
-            // is over
-            acknowledgement_sender: _acknowledgement_sender,
-            ..
-        }) = block_importing_notification_stream.next().await
-        {
-            let block_number_to_archive =
-                match block_number.checked_sub(&confirmation_depth_k.into()) {
-                    Some(block_number_to_archive) => block_number_to_archive,
-                    None => {
-                        continue;
-                    }
-                };
+        'initialization_recovery: loop {
+            let archived_segment_notification_sender =
+                subspace_link.archived_segment_notification_sender.clone();
 
-            if best_archived_block_number >= block_number_to_archive {
-                // This block was already archived, skip
-                continue;
+            let archiver = match maybe_archiver.take() {
+                Some(archiver) => archiver,
+                None => {
+                    initialize_archiver(&segment_headers_store, &subspace_link, client.as_ref())?
+                }
+            };
+
+            let InitializedArchiver {
+                confirmation_depth_k,
+                mut archiver,
+                older_archived_segments,
+                best_archived_block: (mut best_archived_block_hash, mut best_archived_block_number),
+            } = archiver;
+
+            debug!(%best_archived_block_number, ?best_archived_block_hash,  "Archiver initialized.");
```

Comment:
Number is already logged at `info` level in archiver initialization, not sure why this is necessary

Reply:
It seems to be just a duplicated message unless I'm missing something